File: week-01/day-5/src/DrawSquare.java
import java.util.Scanner;


public class DrawSquare {
    public static void main(String[] args) {
        // Write a program that reads a number from the standard input, then draws a
        // square like this:
        //
        //
        // %%%%%%
        // %    %
        // %    %
        // %    %
        // %    %
        // %%%%%%
        //
        // The square should have as many lines as the number was

        // ask the user for the length of the sides
        Scanner scanner = new Scanner(System.in);
        int userInput = scanner.nextInt();

        // 1 or less % sign is not quite a square
        while (userInput < 2) {
            System.out.print("Please type in a higher number! ");
            userInput = scanner.nextInt();
        }

        // iterate through lines
        for (int line = 0; line < userInput; line++) {
            // separate first and last line
            if (line == 0 || line == userInput - 1) {
                for (int i = 0; i < userInput; i++) {
                    System.out.print("%");
                }
            } else {
                System.out.print("%");
                for (int j = 0; j < userInput - 2; j++) {
                    System.out.print(" ");
                }
                System.out.print("%");
            }
            System.out.println();
        }
    }
}

File: python_solutions/week-01/day-4/draw_square.py
# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%%
# %    %
# %    %
# %    %
# %    %
# %%%%%%
#
# The square should have as many lines as the number was

user_input = int(input("numma: "))

while user_input < 1:
    user_input = int(input("bigga numma: "))
else:
    for i in range(1, user_input + 1):
        number_of_percentages = user_input if i in (1, user_input) else 2
        
        if number_of_percentages == user_input:
            print("%" * number_of_percentages)
        else:
            print("%{}%".format(" " * (user_input - number_of_percentages)))
